src/lib.rs

```diff
@@ -361,6 +362,46 @@ impl<F> Server<F> where F: Send + Sync + 'static + Fn(&Request) -> Response {
         }
     }
 
+    /// Creates a new thread for the server that can be gracefully stopped later.
+    /// 
+    /// This function returns a tuple of a `JoinHandle` and a `Sender`.
+    /// You must call `JoinHandle::join()` otherwise the server will not run until completion.
+    /// The server can be stopped at will by sending it an empty `()` message from another thread.
+    /// 
+    /// ```no_run
+    /// use std::thread;
+    /// use std::time::Duration;
+    /// use rouille::Server;
+    /// use rouille::Response;
+    ///
+    /// let server = Server::new("localhost:0", |request| {
+    ///     Response::text("hello world")
+    /// }).unwrap();
+    /// println!("Listening on {:?}", server.server_addr());
+    /// let (handle, sender) = server.stoppable();
+    /// 
+    /// // Stop the server in 3 seconds
+    /// thread::spawn(move || {
+    ///     thread::sleep(Duration::from_secs(3));
+    ///     sender.send(()).unwrap();
+    /// });
+    /// 
+    /// // Block the main thread until the server is stopped
+    /// handle.join().unwrap();
+    /// ```
+    #[inline]
+    pub fn stoppable(self) -> (thread::JoinHandle<()>, mpsc::Sender<()>) {
+        let (tx, rx) = mpsc::channel();
+
+        let handle = thread::spawn(move || {
+            while let Err(_) = rx.try_recv() {
+                self.poll();
+            }
```

Comment:
Won't this thread eat your CPU relentlessly?

Reply:
Yes actually, thank you for pointing that out.